pkg/cmd/openfaas_app_test.go

```diff
@@ -17,3 +22,47 @@ func Test_getValuesSuffix_aarch64(t *testing.T) {
 		t.Errorf("suffix, want: %s, got: %s", want, got)
 	}
 }
+
+var (
+	unexpectedErr         = "failed to merge err: %v, existing flags: %v, set overrides: %v"
+	unexpectedMergeErr    = "error merging flags, expected: %v, got: %v"
+	inconsistentErrString = "inconsistent error return expected: %v, got: %v"
+)
+
+func Test_mergeFlags(t *testing.T) {
+	tests := []struct {
+		flags          map[string]string
+		overrides      []string
+		resultExpected map[string]string
+		errExpected    error
+	}{
+		// positive cases:
+		{map[string]string{}, []string{"b=1"}, map[string]string{"b": "1"}, nil},
+		{map[string]string{}, []string{}, map[string]string{}, nil},
+		{map[string]string{"a": "1"}, []string{}, map[string]string{"a": "1"}, nil},
+		{map[string]string{"a": "1"}, []string{"a=2"}, map[string]string{"a": "2"}, nil},
+		{map[string]string{"a": "1"}, []string{"b=1"}, map[string]string{"a": "1", "b": "1"}, nil},
+		{map[string]string{"a": "1"}, []string{"a=2", "b=1"}, map[string]string{"a": "2", "b": "1"}, nil},
+		{map[string]string{"a": "1", "b": "2"}, []string{"a=2", "b=3"}, map[string]string{"a": "2", "b": "3"}, nil},
+
+		// check errors
+		{map[string]string{"a": "1"}, []string{"a:2"}, nil, fmt.Errorf("incorrect format for custom flag `a:2`")},
+		{map[string]string{"a": "1"}, []string{"a 2"}, nil, fmt.Errorf("incorrect format for custom flag `a 2`")},
+		{map[string]string{"a": "1"}, []string{"a-2"}, nil, fmt.Errorf("incorrect format for custom flag `a-2`")},
+	}
+
+	for _, test := range tests {
```

Comment:
This is a great start.

You'll then use the test name scenario here to add it to the stack

See this example for how it could look: https://github.com/alexellis/derek/blob/master/handler/comment_handler_test.go#L70